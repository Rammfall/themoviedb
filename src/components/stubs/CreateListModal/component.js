import React from 'react'
import { Form, Input } from 'antd'

const CreateListModal = () => (
  <Form>
    <Form.Item
      validateStatus="error"
      help="Should be combination of numbers & alphabets"
    >
      <Input placeholder="Name" />
    </Form.Item>
    <Form.Item>
      <Input placeholder="Description" />
    </Form.Item>
  </Form>
)

export default CreateListModal
