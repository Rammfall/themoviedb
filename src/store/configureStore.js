import { createLogicMiddleware } from 'redux-logic'
import { createStore, applyMiddleware } from 'redux'

import logics from './logics'
import rootReducer from './rootReducer'

const bindMiddleware = (middleware) => {
  if (process.env.NODE_ENV !== 'production') {
    // eslint-disable-next-line
    const { composeWithDevTools } = require('redux-devtools-extension')

    return composeWithDevTools(applyMiddleware(...middleware))
  }

  return applyMiddleware(...middleware)
}

const configureStore = (initialState = {}) => {
  // Add dependencies to pass them to logic functions
  const dependencies = {}

  const logicMiddleware = createLogicMiddleware(logics, dependencies)
  const store = createStore(rootReducer, initialState, bindMiddleware([logicMiddleware]))

  return store
}

export default configureStore
