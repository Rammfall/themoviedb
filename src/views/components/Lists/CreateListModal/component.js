import React from 'react'
import { Modal } from 'antd'
import { useIntl } from 'react-intl'
import { Form, Field } from 'formik'
import PropTypes from 'prop-types'

import InputField from 'Views/components/shared/InputField'

const CreateListModalComponent = ({
  isVisible,
  toggleHandler,
  isSubmitting,
  handleSubmit
}) => {
  const { formatMessage } = useIntl()

  return (
    <Modal
      visible={isVisible}
      onCancel={toggleHandler}
      okText={formatMessage({ id: 'lists.create.button' })}
      title={formatMessage({ id: 'lists.create.title' })}
      confirmLoading={isSubmitting}
      okButtonProps={{ htmlType: 'submit' }}
      onOk={handleSubmit}
    >
      <Form>
        <Field
          name="name"
          component={InputField}
          placeholder={formatMessage({
            id: 'lists.create.name'
          })}
        />
        <Field
          name="description"
          component={InputField}
          placeholder={formatMessage({
            id: 'lists.create.description'
          })}
        />
      </Form>
    </Modal>
  )
}

CreateListModalComponent.propTypes = {
  isVisible: PropTypes.bool.isRequired,
  toggleHandler: PropTypes.func.isRequired,
  isSubmitting: PropTypes.bool.isRequired,
  handleSubmit: PropTypes.func.isRequired
}

export default CreateListModalComponent
