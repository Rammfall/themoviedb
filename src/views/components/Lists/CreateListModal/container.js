import React from 'react'
import { compose } from 'ramda'
import { connect } from 'react-redux'
import { withFormik } from 'formik'
import * as yup from 'yup'
import PropTypes from 'prop-types'
import { withRouter } from 'react-router-dom'

import addListSubmit from 'Utils/formikEvents/addListSubmit'
import { addList } from 'Store/concepts/lists/actions'

import CreateListModalComponent from './component'

const CreateListModal = ({
  isVisible,
  toggleHandler,
  isSubmitting,
  handleSubmit
}) => (
  <CreateListModalComponent
    isVisible={isVisible}
    toggleHandler={toggleHandler}
    isSubmitting={isSubmitting}
    handleSubmit={handleSubmit}
  />
)

CreateListModal.propTypes = {
  isVisible: PropTypes.bool.isRequired,
  toggleHandler: PropTypes.func.isRequired,
  isSubmitting: PropTypes.bool.isRequired,
  handleSubmit: PropTypes.func.isRequired
}

const mapDispatchToProps = {
  onSubmit: addList
}

export { CreateListModal as CreateListModalContainer }
export default compose(
  connect(undefined, mapDispatchToProps),
  withRouter,
  withFormik({
    validationSchema: yup.object().shape({
      name: yup.string().required(),
      description: yup.string().required()
    }),
    mapPropsToValues: () => ({
      name: '',
      description: ''
    }),
    handleSubmit: addListSubmit
  })
)(CreateListModal)
