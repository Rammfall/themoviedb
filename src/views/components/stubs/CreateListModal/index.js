import CreateListModal from './component'

export default CreateListModal
