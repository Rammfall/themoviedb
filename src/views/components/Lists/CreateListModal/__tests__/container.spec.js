import React from 'react'
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'

import diveTo from 'TestUtils/diveToEnzyme'

import CreateListModal, { CreateListModalContainer } from '../container'

jest.mock('react-router-dom', () => ({
  withRouter: component => component
}))

describe('CreateListModal', () => {
  const store = configureStore()()

  it('matches snapshot', () => {
    const props = {
      isVisible: false,
      toggleHandler: jest.fn,
      isSubmitting: false,
      handleSubmit: jest.fn
    }
    const wrapper = shallow(<CreateListModal store={store}
{...props} />)
    const container = diveTo(wrapper, CreateListModalContainer)

    expect(container).toMatchSnapshot()
  })
})
