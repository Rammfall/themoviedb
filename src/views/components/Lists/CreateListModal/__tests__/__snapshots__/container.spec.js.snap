// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateListModal matches snapshot 1`] = `
<CreateListModalComponent
  handleSubmit={[Function]}
  isSubmitting={false}
  isVisible={false}
  toggleHandler={[Function]}
/>
`;
